modules/oe_content_event/tests/src/Kernel/EventLinkSourceFilterTest.php

```diff
@@ -164,6 +177,31 @@ public function testEventLinkSourceFilter(): void {
     ];
     $this->assertEqual($cache->getCacheTags(), $date_cache_tags);
 
+    // Set the time to the end date of an ongoing event.
+    $static_time = new DrupalDateTime('2050-05-15 12:00:00', DateTimeItemInterface::STORAGE_TIMEZONE);
+    $time->setTime($static_time->getTimestamp());
+    $query = $storage->getQuery();
+    // Configuring the filter for upcoming events.
+    $plugin->setConfiguration(['period' => EventPeriodFilter::UPCOMING]);
+    $cache = new CacheableMetadata();
+    $plugin->apply($query, [], $cache);
+    $query_results = $query->execute();
+    $ongoing_events = [
```

Comment:
This test makes no sense, according to the comment we "Set the time to the end date of an ongoing event." (which by the way is also the end date for the upcoming event, which makes things even more confusing), so when we use the "upcoming" filter I would assume that the ongoing event SHOULDN'T be in the results, and still it is (together with the upcoming event which has also ended by now).

Reply:
Yeah, but that is the requirement... :(

Reply:
We have identified a potential conflict with the fact that we are using the cache tags with the exact end date, but actually the end date is still the date of the event that is ongoing and the granularity of the cache is hour based. Ex.: End date today at 12:00:00 - at this exact time the event is still ongoing the event will become past event when the time hits 12:00:01, but the cache gets invalidated before and that causes inconsistencies. We have to discuss this problem with KUGLER Tamas to understand what should be the minimum granularity of the time caching.